server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java

```diff
@@ -0,0 +1,139 @@
+package org.infinispan.server.core.transport;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.management.AttributeNotFoundException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+
+import org.infinispan.Cache;
+import org.infinispan.commons.CacheException;
+import org.infinispan.configuration.global.GlobalConfiguration;
+import org.infinispan.distexec.DefaultExecutorService;
+import org.infinispan.distexec.DistributedCallable;
+import org.infinispan.distexec.DistributedExecutorService;
+import org.infinispan.jmx.JmxUtil;
+import org.infinispan.manager.EmbeddedCacheManager;
+
+import io.netty.channel.group.ChannelGroup;
+
+public class NettyTransportConnectionStats {
+
+   private final EmbeddedCacheManager cacheManager;
+   private final boolean isGlobalStatsEnabled;
+   private final ChannelGroup acceptedChannels;
+   private final String threadNamePrefix;
+   private final AtomicLong totalBytesWritten = new AtomicLong();
+   private final AtomicLong totalBytesRead = new AtomicLong();
+
+   public NettyTransportConnectionStats(EmbeddedCacheManager cacheManager, ChannelGroup acceptedChannels, String threadNamePrefix) {
+      this.cacheManager = cacheManager;
+      this.isGlobalStatsEnabled = cacheManager != null ? cacheManager.getCacheManagerConfiguration().globalJmxStatistics().enabled() : false;
+      this.acceptedChannels = acceptedChannels;
+      this.threadNamePrefix = threadNamePrefix;
+   }
+
+   private void increment(AtomicLong base, long bytes) {
+      if (isGlobalStatsEnabled)
+         base.addAndGet(bytes);
+   }
+
+
+   public void incrementTotalBytesWritten(long bytes) {
+      increment(totalBytesWritten, bytes);
+   }
+
+   public void incrementTotalBytesRead(long bytes) {
+      increment(totalBytesRead, bytes);
+   }
+
+   public long getTotalBytesWritten() {
+      return totalBytesWritten.get();
+   }
+
+   public long getTotalBytesRead() {
+      return totalBytesRead.get();
+   }
+
+   private boolean needDistributedCalculation() {
+      if (cacheManager != null) {
+         org.infinispan.remoting.transport.Transport transport = cacheManager.getTransport();
+         return transport != null && transport.getMembers().size() > 1;
+      }
+      return false;
+   }
+
+   private int calculateGlobalConnections() {
+      Cache<Object, Object> cache = cacheManager.getCache();
+      DistributedExecutorService exec = new DefaultExecutorService(cache);
```

Comment:
I know you simply copied this from `NettyTransport`, but could we use `ClusterExecutor` instead as `DistributedExecutorService` is deprecated?

Reply:
TBH, I don't know. Maybe @wburns will know what's the latest, greatest...

Reply:
Ok, so for now I'll leave it as is... Once I get a clarification from @wburns I'll consider changing this.

Reply:
`DistributedExecutorService` is deprecated, so this is a good time to move it over to `ClusterExecutor` :)

Reply:
Yeah we should move this over to `ClusterExecutor`. You should be able to use something like: https://gist.github.com/wburns/0ebd63d53c9384c3ca021baac70641f6

Reply:
+1, but as I explained on the bottom of the page, I would prefer to implement it in a separate PR. I created ISPN-8873 to track this.